### PR TITLE
Add a few template based sites

### LINKED ISSUE #3277: [Site Request] mangadisk
**Name of the website**
mangadisk

**Website urls (examples below)**
- Site: https://mangadisk.web.id/
- Manga List: https://mangadisk.web.id/manga-lists/
- Manga example: https://mangadisk.web.id/manga/haite-kudasai-takamine-san/
- Chapter online viewer example: https://mangadisk.web.id/haite-kudasai-takamine-san-ch-26-bahasa-indonesia/

**Languages**
Indonesia

### LINKED ISSUE #3603: [Site Request] Bananascan
**Name of the website**
Bananascan

**Website urls (examples below)**
- Site: https://banana-scan.com/
- Manga List: https://banana-scan.com/nos-projets/
- Manga example: https://banana-scan.com/manga/maidens-in-law/
- Chapter online viewer example: https://banana-scan.com/maidens-in-law-chapitre-1/

**Languages**
French

**Website relationship*
None

**Additional details**
template : WordPressMangastream

### LINKED ISSUE #3609: [Site Request] mangapl
**Name of the website**
mangapl

**Website urls (examples below)**
- Site: https://mangapl.com/
- Manga List: https://mangapl.com/?s&post_type=wp-manga&adult=1&m_orderby=new-manga
- Manga example: https://mangapl.com/manga/fitness-uncensored/
- Chapter online viewer example: https://mangapl.com/manga/fitness-uncensored/chapter-10/

**Languages**
List of languages supported by the website

**Website relationship*
If applicable describe the relation with any other website (eg : "alternative domain, copy of ...").

**Additional details**
This Website Had Uncensored Version of Manga

### LINKED ISSUE #3617: [Site Request] pairof2+
**Name of the website**
pairof2+

**Website urls (examples below)**
- Site: https://pairof2.com/
- Manga List: https://pairof2.com/project/
- Manga example: https://pairof2.com/manga/mushoku-tensei-isekai-ittara-honki-dasu/
- Chapter online viewer example: https://pairof2.com/mushoku-tensei-jobless-reincarnation-chapter-72/

**Languages**
English

**Website relationship*
If applicable describe the relation with any other website (eg : "alternative domain, copy of ...").

**Additional details**
Add any other context details that you may have found (like template to reuse)

### PR DESCRIPTION
**Pull request by XenonR**
Closes #3617 (pairoftwo)
Closes #3609 (mangapl)
Closes #3603 (bananascan)
Closes #3277 (mangadisk)

### CODE REVIEW

File: src/web/mjs/connectors/MangaDisk.mjs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    async _getPages(chapter) {
+        const uri = new URL(chapter.id, this.url);
+        let request = new Request(uri, this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryPages);
+
+        return data.map(hash => {
+            const payload = {
+                url: 'https://img.nesia.my.id/image?id='+hash.attributes["file-id"].value,
+                referer: this.url
```
Comment (27136773) [inline src/web/mjs/connectors/MangaDisk.mjs]: The method `_handleConnectorURI()` already has access to `this.url`, so there is no need to provide it via referer in the payload, or was it meant to be `request.url` instead?

File: src/web/mjs/connectors/MangaDisk.mjs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            };
+            return this.createConnectorURI(payload);
+        });
+    }
+
+    async _handleConnectorURI(payload) {
+        let request = new Request(new URL(payload.url), this.requestOptions);
+        request.headers.set('x-referer', payload.referer, this.url);
```
Comment (27136773) [inline src/web/mjs/connectors/MangaDisk.mjs]: The `Headers.set()` method only supports two parameters, the 3rd will be ignored

File: src/web/mjs/connectors/MangaDisk.mjs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    async _getPages(chapter) {
+        const uri = new URL(chapter.id, this.url);
+        let request = new Request(uri, this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryPages);
+
+        return data.map(hash => {
+            const payload = {
+                url: 'https://img.nesia.my.id/image?id='+hash.attributes["file-id"].value,
```
Comment (27136773) [inline src/web/mjs/connectors/MangaDisk.mjs]: May use `hash.getAttribute('file-id')` which will directly return the value

File: src/web/mjs/connectors/MangaDisk.mjs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                referer: this.url
+            };
+            return this.createConnectorURI(payload);
+        });
+    }
+
+    async _handleConnectorURI(payload) {
```
Comment (27136773) [inline src/web/mjs/connectors/MangaDisk.mjs]: If `this.url` is sufficient as `x-referer`, then there is no need to overwrite this method, the base class implementation will cover this when the connector adds the `x-referer` to the `requestOptions` in the constructor,
e.g.
```javascript
this.url = 'https://manga.adonisfansub.com';
this.requestOptions.headers.set('x-referer', this.url);
```

**Review by 27136773 — APPROVED** [review]
Looks good, minor improvements possible

File: src/web/mjs/connectors/MangaDisk.mjs (hunk 1 above)
Comment (XenonR) [reply src/web/mjs/connectors/MangaDisk.mjs]: You are right. Makes no sense.

**Pull request merged**